src/Microsoft.Extensions.Primitives/CompositeChangeToken.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Microsoft.Extensions.Primitives
+{
+    /// <summary>
+    /// An <see cref="IChangeToken"/> which represents one or more <see cref="IChangeToken"/> instances.
+    /// </summary>
+    public class CompositeChangeToken : IChangeToken
+    {
+        private static readonly Action<object> _onChangeDelegate = OnChange;
+        private readonly object _callbackLock = new object();
+        private CancellationTokenSource _cancellationTokenSource;
+        private bool _registeredCallbackProxy;
+        private List<IDisposable> _disposables;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="CompositeChangeToken"/>.
+        /// </summary>
+        /// <param name="changeTokens">The list of <see cref="IChangeToken"/> to compose.</param>
+        public CompositeChangeToken(IReadOnlyList<IChangeToken> changeTokens)
+        {
+            ChangeTokens = changeTokens ?? throw new ArgumentNullException(nameof(changeTokens));
+            for (var i = 0; i < ChangeTokens.Count; i++)
+            {
+                if (ChangeTokens[i].ActiveChangeCallbacks)
+                {
+                    ActiveChangeCallbacks = true;
+                    break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the list of <see cref="IChangeToken"/> which compose the current <see cref="CompositeChangeToken"/>.
+        /// </summary>
+        public IReadOnlyList<IChangeToken> ChangeTokens { get; }
+
+        /// <inheritdoc />
+        public IDisposable RegisterChangeCallback(Action<object> callback, object state)
+        {
+            EnsureCallbacksInitialized();
+            if (!_cancellationTokenSource.IsCancellationRequested)
```

Comment:
Yeah, I think it'd be better if we matched the behavior of `CancellationChangeToken` here and not do any more. We could dispose the token source in `OnChange` so that it throws the `ObjectDisposedException`s if this is invoked. As for the test, we should modify it to match this behavior.